### ISSUE TITLE
FR: keys relative to scope

### ISSUE THREAD

**Issue by wilem82** [lightbend/config#533]
Hi, love your lib, been using on our project for three years.  Only thing that makes configs awkward to work with is when I'm using defaults, like

```
myapp {
    def.url = "http://127.0.0.1"

    blah1 = ${def.url}"/blah1"
    blah2 = ${def.url}"/blah1"
    blah3 = ${def.url}"/blah1"
}
```

Specifying `def.url` relative to `myapp` doesn't work, have to go with the full path `myapp.def.url` which makes for a lot of cumbersome copy-paste, especially in real world, large config files with more nested sections.

Would be great if it worked, perhaps expressed as `${.def.url}` to distinguish absolute from relative.

Any chance this can be done? Thanks.

**Comment by mmuschalik** [comment]
Yes! This is also what I'm after. Being able to use a relative path for substitution would help a lot.

hello.conf:

```
hello {
  msg = 'hello '${../name}
}
```

app.conf:

```
hello_david {
  name = "david"
  include "hello.conf"
}

hello_lisa {
  name = "lisa"
  include "hello.conf"
}
```


Then I'd expect:
`hello_david.hello.msg = "hello david"`

Likewise for Lisa:
`hello_lisa.hello.msg = "hello lisa"`

This enables me to **reuse** conf files with different substitutions.

**Comment by havocp** [comment]
I'm open to it. I think the discussion on https://github.com/lightbend/config/issues/40 is about the same issue. It needs thinking through.
I do think searching global scope first will be a bit surprising here, and even searching lexical scope second maybe isn't quite backward-compatible. So that would argue perhaps for a different syntax for lexical search as suggested on that other issue.
If you agree that #40 is the same we could close this issue to avoid duplication.
Thanks!

**Comment by mmuschalik** [comment]
Yes, I agree the local scope would solve my issue. How about:

Local scope:
`${this.name}`
Global scope:
`${name}`
